### ISSUE TITLE
some question about bbox_head output

### ISSUE THREAD

**Issue by muzi2045** [tianweiy/CenterPoint#30]
trying to export to onnx from prtrained waymo model from CenterPoint codebase (only detection vehicle class):

![image](https://user-images.githubusercontent.com/12984898/90710045-216ba880-e2d0-11ea-9a4b-352da4f73afd.png)

here is the original pointpillars bbox header(9 classes) export from OpenPCDet:
![image](https://user-images.githubusercontent.com/12984898/90710142-511ab080-e2d0-11ea-80a0-f557351e4eba.png)

and the nuscenens bbox header export from CenterPoint (10 classes):
![image](https://user-images.githubusercontent.com/12984898/90710187-6c85bb80-e2d0-11ea-92df-9cce9965ccc7.png)

is there really need to separate the box_head into so many heads with [x,y],[z],[lwh],[yaw_sin, yaw_cos], [class] ?
if here is like 10 classes to be trained, there will be 5*10 heads in bbox_heads with the same feature map, it looks like not a good way to deploy it in a real scene.
@tianweiy

**Comment by tianweiy** [comment]
I agree. I guess combining all regression targets together won't hurt too much (maybe ~~0.5 mAP). I did it this way for easier coding. I will find time to support only two heads (class + all reg) but it will take me a while (probably after this paper gets in somewhere)

**Comment by tianweiy** [comment]
I am working on some major revision of the method and will probably have a large update in the fall.

**Comment by tianweiy** [comment]
```
here is the original pointpillars bbox header(9 classes) export from OpenPCDet:
```

IS this  openpcdet's KITTI PP? I think the recent nuScenes pp still follow the same setup (separate regression into 4 heads)

**Comment by muzi2045** [comment]
Yes, the original kiiti pp config not using multihead.

**Comment by tianweiy** [comment]
Okay, stay tuned. I will get some updates about this later when I get time and GPUs ... I am now mainly exploring the Waymo dataset

**Comment by muzi2045** [comment]
is there any Waymo dataset download mirror link in China? I can't download whole waymo datatset because it's so huge....
@tianweiy

**Comment by tianweiy** [comment]
I am not aware of any and the term disallows redistribution so you may need to ask the team directly...

**Comment by YazanMurhij995** [comment]
@muzi2045 , Could you please provide more information how to display a graph of the model using onnx ? how to provide an input here ?

**Comment by Qjizhi** [comment]
> trying to export to onnx from prtrained waymo model from CenterPoint codebase (only detection vehicle class):
> 
> ![image](https://user-images.githubusercontent.com/12984898/90710045-216ba880-e2d0-11ea-9a4b-352da4f73afd.png)
> 
> here is the original pointpillars bbox header(9 classes) export from OpenPCDet:
> ![image](https://user-images.githubusercontent.com/12984898/90710142-511ab080-e2d0-11ea-80a0-f557351e4eba.png)
> 
> and the nuscenens bbox header export from CenterPoint (10 classes):
> ![image](https://user-images.githubusercontent.com/12984898/90710187-6c85bb80-e2d0-11ea-92df-9cce9965ccc7.png)
> 
> is there really need to separate the box_head into so many heads with [x,y],[z],[lwh],[yaw_sin, yaw_cos], [class] ?
> if here is like 10 classes to be trained, there will be 5*10 heads in bbox_heads with the same feature map, it looks like not a good way to deploy it in a real scene.
> @tianweiy

@muzi2045 , hi could you please provide the exporting onnx code?
@tianweiy do you have any plan to support exporting onnx code?

**Comment by tianweiy** [comment]
I don't know any onnx so I don't have plan to support this in the near future

**Comment by xavidzo** [comment]
hello @muzi2045, maybe do you know why pointpillars has to be exported to onnx separately, the pfe and the detection head?

**Comment by jaydevshelat** [comment]
@muzi2045 @Qjizhi  did you manage to do ONNX export?  Any script/pointers you can give me will be very helpful.  Thank You in advance!

**Comment by tianweiy** [comment]
there is one here https://github.com/CarkusL/CenterPoint but i never tried

**Comment by jaydevshelat** [comment]
Thank You @tianweiy !  I will look at it.
